services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/ReroutesThrottling.java

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2017 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.cache.service;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ReroutesThrottling {
+    private ExtendableTimeWindow extendableTimeWindow;
+
+    private Map<String, String> reroutes = new HashMap<>();
+
+    public ReroutesThrottling(long timeout, long maxTimeWindowSize) {
+        extendableTimeWindow = new ExtendableTimeWindow(timeout, maxTimeWindowSize);
+    }
+
+    /**
+     * This constructor is used only for testing.
+     *
+     * @param extendableTimeWindow the extendable timewindow.
+     */
+    ReroutesThrottling(ExtendableTimeWindow extendableTimeWindow) {
+        this.extendableTimeWindow = extendableTimeWindow;
+    }
+
+    /**
+     * Keeps current reroute request. Only last correlationId will be saved.
+     *
+     * @param flowId the flow ID
+     * @param correlationId the correlation ID
+     */
+    public void putRequest(String flowId, String correlationId) {
+        reroutes.put(flowId, correlationId);
```

Comment:
Would you mind to add some logs here in order to detect the case when we override some requests. In other words it would be useful to see in logs what happened with reroute request in case when we just ignore it here (some requests might be overridden by a newer one)

Reply:
Fixed.